### PR TITLE
feat(remove): Temporally remove the internal miner functionality

### LINKED ISSUE #8180: Remove or comment out zebra internal miner
In https://github.com/ZcashFoundation/zebra/pull/8136 we added an internal miner to zebra that depends on changes made to the ECC `equihash` crates. The problem is that this changes are currently in the form of pull requests, unpublished in crates.io. For this reason the release publishing to crates.io of Zebra is failing as all dependencies must be published. More info in https://github.com/ZcashFoundation/zebra/pull/8165#issuecomment-1902616143

We need to fix this in order to complete the current ongoing release (1.5.1) which was made in github but was not published to crates.io.

We decided to remove the internal miner code as a temporal solution until the needed PRs (https://github.com/zcash/librustzcash/pull/1088 and https://github.com/zcash/librustzcash/pull/1083) are merged and a new version is published. 

This ticket is to:
- [ ] remove the dependency and related code that depends on it.
- [x] open a ticket to restore once stuff gets merged and new version published.

## Follow up work

Create a hotfix release (1.5.2) with the changes done.

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

Close https://github.com/ZcashFoundation/zebra/issues/8180

#### Check before marking the PR as ready for review:
  - [x] Will the PR name make sense to users?
  - [x] Does the PR have a priority label?
  - [ ] Have you added or updated tests?
  - [ ] Is the documentation up to date?

## Solution

- Removed the least amount of code to get rid of the custom equihash dependency.
- Commented the changes as TODOs in the code.
- Created issue to restore.
- Fixed config files needed to make the tests pass.

## Review

Anyone can review, i marked as critical as we are in a merge freeze and it will not get merged otherwise.

### Reviewer Checklist

Check before approving the PR:
  - [ ] Does the PR scope match the ticket?
  - [ ] Are there enough tests to make sure it works? Do the tests cover the PR motivation?
  - [ ] Are all the PR blockers dealt with?
        PR blockers can be dealt with in new tickets or PRs.

_And check the PR Author checklist is complete._

## Follow Up Work

Create a hotfix including this removal of feature in the changelog. https://github.com/ZcashFoundation/zebra/issues/8181

### CODE REVIEW

File: zebra-chain/Cargo.toml [hunk 1]
```diff
@@ -35,12 +35,10 @@ getblocktemplate-rpcs = [
 ]
 
 # Experimental internal miner support
-internal-miner = [
-    # TODO: replace with "equihash/solver" when that feature is merged and released:
-    # https://github.com/zcash/librustzcash/pull/1083
-    # https://github.com/zcash/librustzcash/pull/1088
-    "equihash-solver",
-]
+# TODO: Internal miner feature functionality wass removed at https://github.com/ZcashFoundation/zebra/issues/8180
```
Comment (arya2) [inline zebra-chain/Cargo.toml]:
```suggestion
# TODO: Internal miner feature functionality was removed at https://github.com/ZcashFoundation/zebra/issues/8180
```

**Review by arya2 — DISMISSED** [review]
This looks good, I think we should skip adding the test config until we resolve [this TODO](https://github.com/ZcashFoundation/zebra/blob/issue8180/zebra-rpc/src/config/mining.rs#L61) if `config_tests` passes without it.

**Comment by oxarbitrage** [comment]
test config changes were done because config test were NOT passing without them.

**Comment by oxarbitrage** [comment]
Images were not build at CI. restarting all tests. https://github.com/ZcashFoundation/zebra/actions/runs/7625326629/job/20769382593?pr=8184#step:44:2

**Comment by oxarbitrage** [comment]
> This looks good, I think we should skip adding the test config until we resolve [this TODO](https://github.com/ZcashFoundation/zebra/blob/issue8180/zebra-rpc/src/config/mining.rs#L61) if config_tests passes without it.

You are right, there are more removals needed to make the tests pass with different feature combinations. I pushed a commit to check against CI (https://github.com/ZcashFoundation/zebra/pull/8184/commits/094e3fef3847f10bfc7df25d9168fe55f7e76dc4)

**Review by arya2 — APPROVED** [review]

**Pull request merged**